layers/+lang/haskell/README.org

```diff
@@ -104,20 +104,26 @@ For information about setting up =$PATH=, check out the corresponding section in
 the FAQ (~SPC h SPC $PATH RET~).
 
 ** Choosing a backend
-To choose a default backend set the layer variable =haskell-completion-backend=:
+To choose a backend set the layer variable =haskell-completion-backend=:
 
-#+BEGIN_SRC elisp
-  (haskell :variables haskell-completion-backend 'ghci)
+#+BEGIN_SRC emacs-lisp
+  (setq-default dotspacemacs-configuration-layers
```

Comment:
This line is wrong, the layer declaration for haskell is not part of the auto-completion declaration. Here I think the old version is more clear.

Reply:
I was also confused with this at first, but I believe it is not wrong -> we are just listing layers in dotspacemacs-configuration-layers, first we list auto-completion, then haskell, and haskell has variables. So auto-completion and haskell are here separate elements of the list, haskell is not part of auto-completion declaration. Although, visually it can be deceiving.

I actually copy-pasted this from the old code, line 138, and just changed `intero` into `dante`, so I didn't really change much, I just moved it.

There is sentence below explaining that auto-completion layer is needed for completion to work, this is from old code and I left as is, and I believe that is why auto-completion was listed in the code example.

I do however understand the confusion here hm! What I did not is move the sentence about `auto-completion` layer being required to the top of the section, since it is a requirement and we don't want somebody to miss it, and then we continue with the simple code example that shows how to set haskell backend. Let me know if this seems better.

Reply:
Yep sounds better. The point is that we should never demand syntax checking or auto completion layer this should be a pure user choice. The hooks make sure that the respective setup code is skipped.

Having a hint in the Readme that these are required for certain features does not hurt.The documentation however already mentiones their plugin nature so I would expect that most users are aware of it.